### PR TITLE
Move currency related code to dimension/currency folder

### PR DESCRIPTION
**Pull request by younies**
Move currency related code to dimension/currency folder in order to split the formatting code of currency and units.

### CODE REVIEW

**Comment by robertbastian** [comment]
Why? `icu::dimension::formatter` is not a natural place for `CurrencyFormatter`

**Comment by younies** [comment]
> Why? `icu::dimension::formatter` is not a natural place for `CurrencyFormatter`

@robertbastian , I do not like to have it in the `mode.rs` either. do you suggest a better file name ?

**Comment by robertbastian** [comment]
`dimension::currency::CurrencyFormatter`?

**Comment by younies** [comment]
> `dimension::currency::CurrencyFormatter`?

I have moved all the currency related code to dimension/currency and moved the `CurrencyFormatter` to `formatter.rs`

This will allow separation of the units formatting code and currency formatting code.

File: components/experimental/src/dimension/currency/formatter.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ pub struct CurrencyFormatter {
     options: CurrencyFormatterOptions,
 
     /// Essential data for the currency formatter.
-    essential: DataPayload<super::provider::currency::CurrencyEssentialsV1Marker>,
+    essential: DataPayload<super::super::provider::currency::CurrencyEssentialsV1Marker>,
```
Comment (robertbastian) [inline components/experimental/src/dimension/currency/formatter.rs:30]: nit: import this

**Review by robertbastian — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
The move seems good. Defer to @robertbastian on review

**Review by sffc — APPROVED** [review]

**Pull request merged**